service/service-mock/service-mock-rest/src/main/java/com/castlemock/service/mock/rest/project/converter/openapi/OpenApiRestDefinitionConverter.java

```diff
@@ -0,0 +1,299 @@
+/*
+ * Copyright 2017 Karl Dahlgren
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.castlemock.service.mock.rest.project.converter.openapi;
+
+import com.castlemock.model.core.http.HttpHeader;
+import com.castlemock.model.core.http.HttpMethod;
+import com.castlemock.model.core.utility.file.FileUtility;
+import com.castlemock.model.mock.rest.domain.*;
+import com.castlemock.service.mock.rest.project.converter.AbstractRestDefinitionConverter;
+import io.swagger.parser.OpenAPIParser;
+import io.swagger.v3.oas.models.OpenAPI;
+import io.swagger.v3.oas.models.Operation;
+import io.swagger.v3.oas.models.PathItem;
+import io.swagger.v3.oas.models.headers.Header;
+import io.swagger.v3.oas.models.responses.ApiResponse;
+import io.swagger.v3.oas.models.responses.ApiResponses;
+import io.swagger.v3.parser.core.models.SwaggerParseResult;
+import joptsimple.internal.Strings;
+import org.apache.commons.lang3.ObjectUtils;
+
+import java.io.File;
+import java.util.*;
+
+/**
+ * The {@link OpenApiRestDefinitionConverter} provides OpenAPI V3 related functionality.
+ */
+public class OpenApiRestDefinitionConverter extends AbstractRestDefinitionConverter {
+    /**
+     * The convert method provides the functionality to convert the provided {@link File} into
+     * a list of {@link RestApplication}.
+     *
+     * @param file             The file which will be converted to one or more {@link RestApplication}.
+     * @param generateResponse Will generate a default response if true. No response will be generated if false.
+     * @return A list of {@link RestApplication} based on the provided file.
+     */
+    @Override
+    public List<RestApplication> convert(final File file, final boolean generateResponse) {
+        final String openApiContent = FileUtility.getFileContent(file);
+        SwaggerParseResult result = new OpenAPIParser().readContents(openApiContent, null, null);
```

Comment:
```suggestion
        ParseOptions parseOptions = new ParseOptions();
        parseOptions.setResolve(true);
        parseOptions.setResolveFully(true);
        SwaggerParseResult result = new OpenAPIParser().readContents(openApiContent, null, parseOptions);
```

Add parse options for resolve $ref. Without this, files containing $ref are processed incorrectly (for example, the names of the responses disappear).